### PR TITLE
Patch experiment with continuous_train_and_eval

### LINKED ISSUE #103: Various InvalidArgumentError in Evaluation
The error manifests in multiple ways, always during evaluation and some kind of shape error:

- #102
- #69
- #98
- #101 

Possible causes:
- Some kind of bug in tf.learn evaluate that causes issues with the GPU during eva (this doesn't seem to be a problem with CPU training)
- Mismatch of parameters or training data in training/eval
- Issue with the data reading / input pipeline, e.g. mismatched sequence lengths.

It could be related to these Tensorflow issues:

- https://github.com/tensorflow/tensorflow/issues/7015
- https://github.com/tensorflow/tensorflow/issues/6469

For everyone having this issue, please answer the following to help debug:

- What versions of TensorFlow are you using?
- Can you run on the CPU without error?
- What versions of CUDA are you using?

### PR DESCRIPTION
**Pull request by dennybritz**
Fixes #103

### CODE REVIEW

**Comment by waterbeach** [comment]
I also got an evaluation problem on en-de data. but I have trained successfully. It seems it can train, but when it is time to do evaluation, it will be failed. Error info is shown below txt file:
[eva.txt](https://github.com/google/seq2seq/files/986888/eva.txt)

**Comment by ghost** [comment]
Is it possible that this training schedule affects the performance of the system?

When I used the previous repo, before this fix, I managed to get a BLEU around 15 with a small model (200.000 training sentences, 2000 dev set, 2000 test set, batch size 32, 178.500 training steps). Whenever I have tried to replicate this experiment with the exact same configurations afterwards, I have only managed to get something around 5 BLEU. The only difference in configuration, is this new training schedule..
